src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/ISearchValue.cs

```diff
@@ -17,6 +17,30 @@ public interface ISearchValue : IEquatable<ISearchValue>
         /// </summary>
         bool IsValidAsCompositeComponent { get; }
 
+        /// <summary>
+        /// Determines whether this current value is the minimum when compared to
+        /// a collection of search values for the same parameter.
+        /// </summary>
+        bool IsMin { get; set; }
+
+        /// <summary>
+        /// Determines whether this current value is the maximum when compared to
+        /// a collection of search values for the same paramter.
+        /// </summary>
+        bool IsMax { get; set; }
+
+        /// <summary>
+        /// Compares two ISearchValue objects of the same type and returns an integer that
+        /// indicates their relative position in the sort order.
+        /// </summary>
+        /// <param name="otherValue">The value to compare against.</param>
+        /// <returns>
+        /// -1 if the current value comes before the given search value.
+        /// 0 if both values will occur at the same position.
+        /// 1 if the current value comes after the given search value.
+        /// </returns>
+        int Compare(ISearchValue otherValue);
```

Comment:
Should we update this to IComparable interface? and rename to CompareTo?

Reply:
@feordin  I tried this out after you mentioned it. One issue is that we not only have to implement `CompareTo`, but also override all equality operations and `GetHashCode()`. We won't be using some of that code (at least not right now). So I am thinking it might be okay to have `Compare` as a standalone method. And if things change in the future we can change the code to implement `IComparable`. Thoughts?

Reply:
>but also override all equality operations and GetHashCode()
Why tho? IComparable<T> require you to only provide CompareTo method.

Reply:
Yes, the interface only asks for `CompareTo`. But I ran into the following warning when doing so - [CA1036](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1036?view=vs-2019).

Reply:
Well, no one can stop you from disabling that warning, as long as you know what you doing...

Reply:
I usually prefer to avoid suppressing warnings. I have updated the class to implement `IComparable` and suppressed the warning.